tools/cli/go-whisk-cli/commands/qualified_name.go

```diff
@@ -0,0 +1,207 @@
+    output := []string{}
+
+    if len(qualifiedName.namespace) > 0 {
+        output = append(output, "/", qualifiedName.namespace, "/")
+    }
+    if len(qualifiedName.packageName) > 0 {
+        output = append(output, qualifiedName.packageName, "/")
+    }
+    output = append(output, qualifiedName.entity)
+
+    return strings.Join(output, "")
+}
+
+//  GetPackageName() returns the package name from qualifiedName without a
+//      leading '/'
+func (qualifiedName *QualifiedName) GetPackageName() string {
+    return qualifiedName.packageName
+}
+
+//  GetEntityName() returns the entity name ([package/]entity) of qualifiedName
+//      without a leading '/'
+func (qualifiedName *QualifiedName) GetEntityName() string {
+    return qualifiedName.entityName
+}
+
+//  GetEntity() returns the name of entity in qualifiedName without a leading '/'
+func (qualifiedName *QualifiedName) GetEntity() string {
+    return qualifiedName.entity
+}
+
+//  GetNamespace() returns the name of the namespace in qualifiedName without
+//      a leading '/'
+func (qualifiedName *QualifiedName) GetNamespace() string {
+    return qualifiedName.namespace
+}
+
+//  addLeadSlash(name) returns a (possibly fully qualified) resource name,
+//      inserting a leading '/' if it is of 3 parts (namespace/package/action)
+//      and lacking the leading '/'.
+func addLeadSlash(name string) string {
+    parts := strings.Split(name, "/")
+    if len(parts) == 3 && parts[0] != "" {
+        name = "/" + name
+    }
+    return name
+}
+
+//  NewQualifiedName(name) initializes and constructs a (possibly fully qualified)
+//      QualifiedName struct.
+//
+//      NOTE: If the given qualified name is None, then this is a default qualified
+//          name and it is resolved from properties.
+//      NOTE: If the namespace is missing from the qualified name, the namespace
+//          is also resolved from the property file.
+//
+//  Examples:
+//      foo => qualifiedName {namespace: "_", entityName: foo}
+//      pkg/foo => qualifiedName {namespace: "_", entityName: pkg/foo}
+//      /ns/foo => qualifiedName {namespace: ns, entityName: foo}
+//      /ns/pkg/foo => qualifiedName {namespace: ns, entityName: pkg/foo}
+func NewQualifiedName(name string) (*QualifiedName, error) {
+    qualifiedName := new(QualifiedName)
+
+    // If name has a preceding delimiter (/), or if it has two delimiters with a
+    // leading non-empty string, then it contains a namespace. Otherwise the name
+    // does not specify a namespace, so default the namespace to the namespace
+    // value set in the properties file; if that is not set, use "_"
+    name = addLeadSlash(name)
+    parts := strings.Split(name, "/")
+    if  strings.HasPrefix(name, "/")  {
+        qualifiedName.namespace = parts[1]
+
+        if len(parts) < 2 || len(parts) > 4 {
+            whisk.Debug(whisk.DbgError, "A valid qualified name was not detected\n")
```

Comment:
This code is duplicated below. Can move the body of the if block to a function and just call that where needed.